wallet/wallet.go

```diff
@@ -657,11 +657,25 @@ func (w *Wallet) syncWithChain() error {
 	// as well.  I am leaning towards allowing off all rpcclient
 	// notification re-registrations, in which case the code here should be
 	// left as is.
-	err = chainClient.NotifyBlocks()
-	if err != nil {
+	if err := chainClient.NotifyBlocks(); err != nil {
 		return err
 	}
 
+	// If this was our initial sync, then a birthday stamp should have been
+	// populated, so we'll write it to disk now.
+	if birthdayStamp != nil {
+		log.Debugf("Using birthday block: height=%d, hash=%v",
+			birthdayStamp.Height, birthdayStamp.Hash)
+
+		err := walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
+			ns := tx.ReadWriteBucket(waddrmgrNamespaceKey)
+			return w.Manager.SetBirthdayBlock(ns, *birthdayStamp)
```

Comment:
It's already set at the time the wallet is created. This is the only reasonable place to set the birthday block because the wallet is not aware of the chain yet (in most scenarios) when it is created. For example, in `lnd`, we create the wallet before we even establish our connection to the backend, so at this point we wouldn't be able to retrieve our birthday block.

Reply:
> Why not set this when the birthday block is discovered? Right now it seems we may never set it if we crash after finding it but before finishing the initial sync.

The birthday block can be modified later on by a rollback, which is why I opted for doing it here instead. Should we just also persist it once we find it as well?

> It should also be loaded at the beginning of this method in case it was written in a prior attempt

Why does this need to be done though?

Reply:
This should be moved above to when we discover the birthday, such that, it's written in the very same transaction as the call to `SyncedTo` directly after we discover the birthday. Otherwise, we can write the `SyncedTo` as the birthday height, then crash _right_ after the call to `Commit()`. If this happens, then we'll come back up again, and note the birthday stamp to be nil, and potentially write (at the bottom) an incorrect birthday height _right_ after the _actual_ birthday height. 

This won't be caught by the sanity check as no height has been written yet.

Reply:
Fixed. Let me know if there's anything I missed.